### DISCUSSION TITLE
How to Implement Soft Deletes with Prisma and Cascade Delete Relations?

### DISCUSSION THREAD

**Discussion by Stormer0528** [prisma/prisma#25366] [Q&A] [answered]
### Question

 I'm working with Prisma in my application and need to implement soft deletes for certain models, meaning that records should be marked as deleted instead of being physically removed. However, I also have related models with cascade delete behavior, and I want to ensure that related records follow the soft delete pattern instead of being deleted from the database. What’s the best way to implement this in Prisma?

**Comment by skyward06** [comment, marked as answer]
Prisma doesn’t natively support soft deletes, but you can implement them using a combination of nullable deletedAt fields and middleware to simulate soft deletes. Here's an advanced approach to handle soft deletes, along with cascading soft deletes for related models.

`model User {
  id        Int       @id @default(autoincrement())
  name      String
  posts     Post[]
  deletedAt DateTime? @default(null)
}

model Post {
  id        Int       @id @default(autoincrement())
  title     String
  author    User?     @relation(fields: [authorId], references: [id])
  authorId  Int?
  deletedAt DateTime? @default(null)
}
`
